lib/phoenix_live_view/async.ex

```diff
@@ -54,18 +114,20 @@ defmodule Phoenix.LiveView.Async do
     |> run_async_task(keys, wrapped_func, :assign, opts)
   end
 
-  defp run_async_task(%Socket{} = socket, key, func, kind, opts) do
+  def run_async_task(%Socket{} = socket, key, func, kind, opts) do
```

Comment:
We probably want to add the guard here?

```suggestion
  def run_async_task(%Socket{} = socket, key, func, kind, opts) when is_function(func, 0) do
```